WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/logic/interval/IDEIntervalManagerBase.java

```diff
@@ -0,0 +1,170 @@
+package nl.tudelft.watchdog.core.logic.interval;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.commons.lang.RandomStringUtils;
+
+import nl.tudelft.watchdog.core.logic.interval.intervaltypes.EditorIntervalBase;
+import nl.tudelft.watchdog.core.logic.interval.intervaltypes.IntervalBase;
+import nl.tudelft.watchdog.core.logic.interval.intervaltypes.TypingInterval;
+import nl.tudelft.watchdog.core.util.WatchDogLogger;
+
+/**
+ * Adds functionality for adding and removing intervals.
+ */
+public abstract class IDEIntervalManagerBase extends IntervalManagerBase {
```

Comment:
I thought of that as well. However, IntervalStatistics also extends IntervalManagerBase. So, when we would merge IDEIntervalManagerBase and IntervalManagerBase, IntervalStatistics would suddenly also contain much more functionality (that it doesn't need). Therefore, I haven't merged these two classes, but I can still do it if you like.